lib/lte_link_control/lte_lc.c

```diff
@@ -175,6 +188,51 @@ static void lte_lc_psm_get_work_fn(struct k_work *work_item)
 	lte_lc_evt_psm_update_send(&psm_cfg);
 }
 
+static void lte_lc_edrx_values_store(
+	enum lte_lc_lte_mode mode,
+	char *edrx_value,
+	char *ptw_value)
+{
+	switch (mode) {
+	case LTE_LC_LTE_MODE_LTEM:
+		strcpy(edrx_value_ltem, edrx_value);
+		strcpy(ptw_value_ltem, ptw_value);
+		break;
+	case LTE_LC_LTE_MODE_NBIOT:
+		strcpy(edrx_value_nbiot, edrx_value);
+		strcpy(ptw_value_nbiot, ptw_value);
+		break;
+	default:
+		memset(edrx_value_ltem, 0, sizeof(edrx_value_ltem));
+		memset(ptw_value_ltem, 0, sizeof(ptw_value_ltem));
```

Comment:
Hmm, this should probably also clear `edrx_value_nbiot` and `ptw_value_nbiot`?

Now that I think about it, these values can only be valid for one RAT/LTE mode at a time. So, when mode is LTE-M, should the NB-IoT values be cleared? And vice versa. When switching between modes, the current implementation keeps used values for the old mode. But, we don't know if the same values will be used if modem switches back to that mode. I'm not sure how modem works in this case, though. For example, does it send a new +CEDRXP notification for LTE-M when switching from LTE-M -> NB-IoT -> LTE-M, if the eDRX parameters in LTE-M remain the same?

Reply:
> Hmm, this should probably also clear edrx_value_nbiot and ptw_value_nbiot?

Done